src/Symfony/Component/Routing/Attribute/Route.php

```diff
@@ -33,22 +33,23 @@ class Route
-     * @param string|string[]|null                              $env          The env(s) in which the route is defined (i.e. "dev", "test", "prod", ["dev", "test"])
-     * @param string|DeprecatedAlias|(string|DeprecatedAlias)[] $alias        The list of aliases for this route
+     * @param string|array<string,string>|null                        $path             The route path (i.e. "/user/login")
+     * @param string|null                                             $name             The route name (i.e. "app_user_login")
+     * @param array<string|\Stringable>                               $requirements     Requirements for the route attributes, @see https://symfony.com/doc/current/routing.html#parameters-validation
+     * @param array<string, mixed>                                    $options          Options for the route (i.e. ['prefix' => '/api'])
+     * @param array<string, mixed>                                    $defaults         Default values for the route attributes and query parameters
+     * @param string|null                                             $host             The host for which this route should be active (i.e. "localhost")
+     * @param string|string[]                                         $methods          The list of HTTP methods allowed by this route
+     * @param string|string[]                                         $schemes          The list of schemes allowed by this route (i.e. "https")
+     * @param string|null                                             $condition        An expression that must evaluate to true for the route to be matched, @see https://symfony.com/doc/current/routing.html#matching-expressions
+     * @param int|null                                                $priority         The priority of the route if multiple ones are defined for the same path
+     * @param string|null                                             $locale           The locale accepted by the route
+     * @param string|null                                             $format           The format returned by the route (i.e. "json", "xml")
+     * @param bool|null                                               $utf8             Whether the route accepts UTF-8 in its parameters
+     * @param bool|null                                               $stateless        Whether the route is defined as stateless or stateful, @see https://symfony.com/doc/current/routing.html#stateless-routes
+     * @param string|string[]|null                                    $env              The env(s) in which the route is defined (i.e. "dev", "test", "prod", ["dev", "test"])
+     * @param string|DeprecatedAlias|(string|DeprecatedAlias)[]       $alias            The list of aliases for this route
+     * @param bool|array{params?: string|iterable<array<mixed>>}|null $staticGeneration The static generation configuration, params : route parameters
```

Comment:
we already have `$defaults` for defining default route parameters
and we also have $stateless for identifying routes that could trigger static page generation
can't we rely on those? that'd reduce complexity IMHO

Reply:
A page being stateless is a requirement for being able to make it static, but it might not be enough (it can depend on dynamic data).
Static site generation implies being able to cache the response until the next rebuild.